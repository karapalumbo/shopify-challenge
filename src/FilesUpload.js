import React, { useState, useEffect } from "react";
import axios from "axios";
import Loading from "./Loading";
import { FileCard } from "./FileCard";
import FilesApi from "./api";
import "./Files.css";

export const FilesUpload = () => {
  const [inputValue, setInputValue] = useState("");
  const [files, setFiles] = useState([]);

  axios.defaults.headers.common = {
    "Content-Type": "application/json",
  };

  const handleChange = (event) => {
    event.preventDefault();
    setInputValue(event.target.files[0]);
  };

  const handleRequest = async () => {
    let allFiles = await FilesApi.allFiles();

    let f = allFiles.map((image) => {
      let str = image.split("/");
      let img = str.pop();
      return img;
    });
    setFiles(f);
  };

  useEffect(() => {
    handleRequest();
  }, []);

  const handleUpload = async (e) => {
    e.preventDefault();
    const formData = new FormData();
    const imageFile = inputValue;
    formData.append("image", imageFile);

    await FilesApi.addFiles(formData)
      // await axios
      //   .post("http://localhost:8000/image", formData)
      .then((response) => {
        setFiles([response.data.file.filename, ...files]);
      });
  };

  if (!files) return <Loading />;

  return (
    <>
      <div className="container">
        <div className="row">
          <form>
            <h3>File Upload</h3>
            <div className="form-group">
              <input type="file" onChange={handleChange}></input>
            </div>
            <div className="form-group">
              <button type="submit" onClick={handleUpload}>
                Upload
              </button>
            </div>
          </form>
        </div>
        {files.length ? (
          <div className="container">
            {files.map((file) => {
              return <FileCard filename={file} />;
            })}
          </div>
        ) : (
          <p>Upload a photo!</p>
        )}
      </div>
    </>
  );
};
