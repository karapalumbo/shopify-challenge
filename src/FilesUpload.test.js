import React from "react";
import { render } from "@testing-library/react";
import { FilesUpload } from "./FilesUpload";

it("renders without crashing", function () {
  render(<FilesUpload />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<FilesUpload />);
  expect(asFragment()).toMatchSnapshot();
});
