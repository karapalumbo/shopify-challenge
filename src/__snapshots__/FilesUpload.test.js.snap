// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      class="upload-form"
    >
      <div
        class="form-container"
      >
        <h2>
          Your inventory
        </h2>
        <div
          class="upload-actions"
        >
          <div
            class="form-group"
          >
            <input
              class="css-1jyn810"
              type="file"
            />
          </div>
          <div
            class="form-group"
          >
            <button
              class="css-1hi3plg"
              style="margin-top: 15px;"
              type="submit"
            >
              <span
                class="css-hrreuv"
              >
                Upload
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
    <div
      class="empty-state"
    >
      <p>
        You currently have no images uploaded.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
